src/app/components/elements/OutgoingDelegations.jsx

```diff
@@ -0,0 +1,439 @@
+                        const delegateeA = a.delegatee.toLowerCase();
+                        const delegateeB = b.delegatee.toLowerCase();
+                        if (delegateeA < delegateeB) {
+                            return -1;
+                        }
+                        if (delegateeA > delegateeB) {
+                            return 1;
+                        }
+                        return 0;
+                    })
+                    : data.sort((a, b) => {
+                        const delegateeA = a.delegatee.toLowerCase();
+                        const delegateeB = b.delegatee.toLowerCase();
+                        if (delegateeA < delegateeB) {
+                            return 1;
+                        }
+                        if (delegateeA > delegateeB) {
+                            return -1;
+                        }
+                        return 0;
+                    })
+                break;
+            case DATE:
+                data = sortMethod === ASC
+                    ? data.sort((a, b) => {
+                        const dateA = new Date(a.min_delegation_time).getTime();
+                        const dateB = new Date(b.min_delegation_time).getTime();
+                        return dateA > dateB ? 1 : -1
+                    })
+                    : data.sort((a, b) => {
+                        const dateA = new Date(a.min_delegation_time).getTime();
+                        const dateB = new Date(b.min_delegation_time).getTime();
+                        return dateA > dateB ? -1 : 1
+                    })
+                break;
+            default:
+                break;
+        }
+        this.setState({ sortBy: field, sort: sortMethod, currentPage: 1 })
+        this.updateVestingDelegations(data);
+    }
+
+    handleResize = () => {
+        this.setState({ isSmallScreen: window.innerWidth <= 425 });
+    };
+
+    updateVestingDelegations = (res) => {
+        const {
+            setVestingDelegations,
+            vestingDelegationsLoading,
+        } = this.props;
+        setVestingDelegations(res);
+        vestingDelegationsLoading(false);
+    }
+
+    handleFindAccounts = (e) => {
+        const { value } = e.target;
+        const { vestingDelegationsLoading } = this.props;
+        const { auxiliaryData, currentPage } = this.state;
+        const currentValue = value.replace(/\s/g, '');
+        this.setState({ delegatee: currentValue, sortBy: '', sort: DESC });
+        vestingDelegationsLoading(true);
+        let data = auxiliaryData
+        data = data.filter(item => item.delegatee.includes(currentValue.toLowerCase()))
+        if (currentPage !== 1) {
+            this.setState({ currentPage: 1 })
+        }
+        this.updateVestingDelegations(data)
+    }
+
+    onPageChanged = dt => {
+        this.setState({ currentPage: dt.currentPage })
+    }
+
+    getVestingDelegationsRecursive = (result, timeout) => {
```

Comment:
Using async/await and better error handling could make the code more modern and robust. For example:

```js
getVestingDelegationsRecursive = async (result, timeout) => {
    try {
        const delegatee = result[result.length - 1].delegatee;
        const res = await fetchData(['steem', delegatee, API_LIMIT]);

        if (res.length > 1 && this.isComponentMounted) {
            result = result.concat(res.slice(1));
            if (res.length < API_LIMIT) {
                return result;
            }

            await new Promise((resolve) => setTimeout(resolve, timeout));

            if (this.isComponentMounted) {
                this.setState({ count: result.length });
                const newTimeout = timeout + 250 > 3000 ? 1000 : timeout + 250;
                return this.getVestingDelegationsRecursive(result, newTimeout);
            }
        }

        return result;
    } catch (error) {
        console.log(error);
        return result;
    }
};
```

Reply:
Fix pushed.